### ISSUE TITLE
Make static refactoring (CA1822) doesn't remove "this." references from callers

### ISSUE THREAD

**Issue by mikegoatly** [dotnet/roslyn-analyzers#3709]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

[CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822?view=vs-2019)

#### Repro steps

1. Create the following class:
``` csharp
    class Class1
    {
        public void InstanceMethod()
        {
            this.DoSomething();
        }

        public void DoSomething()
        {
        }
    }
```
2. Install the Microsoft.CodeAnalysis.FxCopAnalyzers package
3. Use the Make Static refactoring on `DoSomething`
4. Code is left in an uncompilable state as the `this.` reference is not removed from the call to `DoSomething`. 
``` csharp
class Class1
    {
        public void InstanceMethod()
        {
            this.DoSomething();
        }

        public static void DoSomething()
        {
        }
    }
```
#### Expected behavior
When making a method static, any references to it should have the `this.` reference removed.

#### Actual behavior
`this.` references remain and the code is left in an uncompilable state.

**Comment by Evangelink** [comment]
Hello @mikegoatly,

I have added the following unit test on the latest master and I cannot reproduce the issue.

```csharp
[Fact, WorkItem(3709, "https://github.com/dotnet/roslyn-analyzers/issues/3709")]
public async Task RemoveThisMemberAccess()
{
	await VerifyCS.VerifyCodeFixAsync(@"
public class Class1
{
    private int field;

    public void InstanceMethod()
    {
        this.DoSomething();
        field++;
    }

    public void [|DoSomething|]()
    {
    }
}", @"
public class Class1
{
    private int field;

    public void InstanceMethod()
    {
        DoSomething();
        field++;
    }

    public static void DoSomething()
    {
    }
}");
}
```

Could you check if the code is correctly reproducing your issue? It's possible that this was fixed in between but I am not sure.

**Comment by mikegoatly** [comment]
@Evangelink thanks for taking a look. 

I've just installed the latest prerelease (3.3.0-beta2.final) and can confirm that it's fixed there so I'll close this issue.

As an aside, I'd not seen how the code fixes are unit tested - that's really slick!

**Comment by Evangelink** [comment]
Thank you for your prompt feedback!
